### DISCUSSION TITLE
issue: In Controller the field.value does not preserve the same object passed to field.onChange

### DISCUSSION THREAD

**Discussion by croraf** [react-hook-form/react-hook-form#11032] [Q&A]
### Version Number

7.47.0

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-v7-controller-ts-forked-lnds5s?file=/src/MuiAutoComplete/MuiAutoComplete.tsx

### Steps to reproduce

1. Find _MUI autocomplete_ field in the codesandbox demo
2. Click on an item from the autocomplete dropdown (some country)
3. Observe the console log that I added in the `Controller` of `CountrySelect` component
4. You can notice that `findIndex` returns -1 with reference comparison of the selected item and items in the options (countries) array.

### Expected behaviour

"Stability" should be preserved, that is what is retrieved from `field.value` should be the same reference passed to the `field.onChange` method.

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Edge

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
`controller` value only specific to its component. You can host your own state with `useState` at the moment.

**Reply by croraf** [reply]
What do you mean by "specific to its component"?
